### PR TITLE
Manifest dt

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: core/arch/arm/kernel/boot.c [hunk 1]
```diff
@@ -1165,12 +1171,18 @@ static bool cpu_nmfi_enabled(void)
  * the unpaged area.
  */
 void __weak boot_init_primary_late(unsigned long fdt,
-				   unsigned long tos_fw_config)
+				   unsigned long manifest_fdt)
```
Comment (etienne-lms) [inline core/arch/arm/kernel/boot.c:1174]: This local `manifest_fdt` and the global `manifest_dt` quite look alike, it a bit confusing.

**Review by etienne-lms — COMMENTED** [review]
The second commit does many things. I felt a bit lost in the implication of requiring a manifest DTB in all the SPMC configurations. Is it always read-only, even when `CFG_CORE_SEL1_SPMC=y`?

File: core/mm/core_mmu.c [hunk 2]
```diff
@@ -755,6 +756,8 @@ uint32_t core_mmu_type_to_attr(enum teecore_memtypes t)
 	case MEM_AREA_NSEC_SHM:
 	case MEM_AREA_NEX_NSEC_SHM:
 		return attr | TEE_MATTR_PRW | cached;
+	case MEM_AREA_MANIFEST_DT:
+		return attr | TEE_MATTR_SECURE | TEE_MATTR_PR;
```
[the remark this answers is not in the document]
Comment (jenswi-linaro) [reply core/mm/core_mmu.c:761]: Something is missing, it should be `cached` since we have coherent memory when using this. I'll fix.

**Comment by jenswi-linaro** [comment]
> The second commit does many things. I felt a bit lost in the implication of requiring a manifest DTB in all the SPMC configurations. Is it always read-only, even when `CFG_CORE_SEL1_SPMC=y`?

I'm sorry, I'll try to break it into smaller pieces. Yes, the manifest is always supposed to be read-only, it's not to be confused with the system DTB.

File: core/arch/arm/kernel/boot.c (hunk 1 above)
Comment (jenswi-linaro) [reply core/arch/arm/kernel/boot.c:1174]: I'll think of something.

**Comment by jenswi-linaro** [comment]
I've split "core: rename tos_fw_config_dt to manifest_dt" into multiple commits.
I've also added "libutils: add runtime_assert()" for better stubbing of `core_mmu_set_secure_memory()` depending on configuration.

**Comment by LeiSen62** [comment]
Acked-by: Leisen <leisen1@huawei.com>

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>` for the series.

**Comment by jenswi-linaro** [comment]
Tags applied.

**Comment by balint-dobszay-arm** [comment]
Hi Jens,
I get a crash when `CFG_CORE_SEL1_SPMC=y` and `CFG_CORE_TPM_EVENT_LOG=y`. After your changes the manifest DT is mapped as RO: https://github.com/OP-TEE/optee_os/pull/6289/commits/aca027763ffaf76c8a36ae787a085b402d212185#diff-e50fbd95497200fea2a8eb3e91c406c07bc915bd20bdc6d76b88bd7c06ee1a74R760

But the TPM event log handling tries to write to it here: https://github.com/OP-TEE/optee_os/blob/master/core/kernel/tpm.c#L50

Making the region RW and/or removing the mentioned `fdt_setprop()` solves the issue. When the DT is in S memory I don't really see a reason to explicitly zero the event log address, so probably removing the write is the better solution.
What do you think?

**Comment by jenswi-linaro** [comment]
I agree, removing the writing makes sense. Do you mind making a PR to fix it or shall I do it?

**Comment by balint-dobszay-arm** [comment]
I'll create a PR. It looks like this only affects the S-EL1 SPMC use case, since otherwise `tpm_map_log_area()` operates on the external DT which is RW.

**Comment by jenswi-linaro** [comment]
Thank you

**Comment by balint-dobszay-arm** [comment]
FYI https://github.com/OP-TEE/optee_os/pull/6310

**Pull request merged**